src/js/dom-utils.js

```diff
@@ -417,6 +447,8 @@ export {
     getXPath,
     hasPreviousCommentSiblingWithContent,
     hasPreviousSiblingElementSameName,
+    hasNextSiblingElementSameName,
+    hasSiblingElementSameName,
```

Comment:
dom-utils is meant to replace jQuery but only with the stuff we use. It's funny, I actually thought I remembered you liked exporting everything and hadn't done this initially.

These new functions are not planned to be used by any apps I know, but I think they could potentially be useful. Not sure about any negative implication of exposing them, but totally fine with me to not export them. They do have some nice tests though....

Reply:
> dom-utils is meant to replace jQuery but only with the stuff we use.

Ah, like every project eventually implements a subset of ~~lisp~~ lodash 🙂 

> I actually thought I remembered you liked exporting everything and hadn't done this initially.

> These new functions are not planned to be used by any apps I know, but I think they could potentially be useful. Not sure about any negative implication of exposing them, but totally fine with me to not export them. They do have some nice tests though....

I do tend to export pretty liberally[^1]! My focus was more on whether `hasNextSiblingElementSameName`/`hasSiblingElementSameName` tests were intended for those functions' direct behavior, or intended for `getXPath` and testing at a lower level. If the former, I would tend to test `getXPath` to exercise the behavior of these functions (which you also did 👍 ) because at that level tends to help associate the test/descriptions with the bug/feature/etc they're intended to address. But since you anticipate them being useful outside of `getXPath` I think this is fine.

[^1]: And we could use `@private` JSDoc tags where we get value from exporting where we don't want to make an API explicit, though of course that doesn't actually prevent it from being used as if it's part of the public API. (Hey look, using the new footnote feature on day 1!)

Reply:
> Hey look, using the new footnote feature on day 1

That's a really cool achievement! 🗡️ I am desperately trying to find a good reason to do this.

> we could use @private JSDoc tags

Yes, that seems useful. How do you like the convention of prepending those functions and properties with `_`, which is done in parts of Enketo (of course not consistently!!).

Thanks for clarifying the testing for getXPath. It makes sense in case the implementation changes. I'll add some tricky comment nodes to those tests now.

Thanks!